### PR TITLE
Initial implementation of GPUBindGroup for WebGPU

### PR DESCRIPTION
**Pull request by imiklos**
Added WebIDL bindings for `GPUBindGroup`.
Implemented the `createBindGroup` function of `GPUDevice`
Renamed `GPUBindGroupBinding` to `GPUBindGroupBindings` and `GPUBufferBinding` to `GPUBufferBindings` in the WebIDL, because these names are already occupied.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes addresses a part of #24706 

cc @kvark @jdm @zakorgy

### CODE REVIEW

File: components/script/dom/identityhub.rs [hunk 1]
```diff
@@ -132,6 +138,20 @@ impl Identities {
         }
     }
 
+    pub fn create_bind_group_id(&mut self, backend: Backend) -> BindGroupId {
```
Comment (kvark) [inline components/script/dom/identityhub.rs]: Why aren't we doing this simpler, like in https://searchfox.org/mozilla-central/rev/ad6234148892bc29bf08d736604ac71925138040/gfx/wgpu/wgpu-remote/src/lib.rs#277

Comment (imiklos) [reply components/script/dom/identityhub.rs]: Yes, you're right! This could be simpler. I will remove the code duplication in another pull request.

Comment (imiklos) [reply components/script/dom/identityhub.rs]: I've just created a PR for this: #25648

File: components/script/dom/gpudevice.rs [hunk 2]
```diff
@@ -463,4 +468,64 @@ impl GPUDeviceMethods for GPUDevice {
...
+            let buffer_size = bind.resource.buffer.size();
+            let resource_size = bind.resource.size.unwrap_or(buffer_size);
+            let length = bind.resource.offset.checked_add(resource_size);
+
+            length.is_some() &&
+            buffer_size >= length.unwrap() && // check buffer OOB
+            bind.resource.offset % alignment == 0 && // check alignment
+            bind.resource.offset < buffer_size && // on Vulkan offset must be less than size of buffer
+            descriptor.layout.bindings().iter().any(|layout_bind| {
+                let usage = BufferUsage::from_bits(bind.resource.buffer.usage()).unwrap();
```
Comment (jdm) [inline components/script/dom/gpudevice.rs]: This could be moved outside of the `any` call, right?

File: components/script/dom/gpudevice.rs [hunk 3]
```diff
@@ -463,4 +468,64 @@ impl GPUDeviceMethods for GPUDevice {
...
+                let ty = match layout_bind.type_ {
+                    GPUBindingType::Storage_buffer  => BufferUsage::STORAGE,
+                    // GPUBindingType::Readonly_storage_buffer  => BufferUsage::STORAGE_READ,
```
Comment (jdm) [inline components/script/dom/gpudevice.rs]: Why is this commented out?

**Review by jdm — APPROVED** [review]
Looks reasonable!

File: components/script/dom/gpudevice.rs (hunk 3 above)
Comment (imiklos) [reply components/script/dom/gpudevice.rs]: The `BufferUsage::STORAGE_READ` is not in the Draft so neither in the WebIDL yet, but it is in wgpu-core. I just wanted it to be in place already, but I can remove it.

File: components/script/dom/gpudevice.rs (hunk 2 above)
Comment (imiklos) [reply components/script/dom/gpudevice.rs]: Yes, you are right! I will move outside the `any`.

**Pull request merged**